evennia/contrib/auditing.py

```diff
@@ -0,0 +1,167 @@
+"""
+
+Contribution - Johnny 2017
+
+This is an extension of the base ServerSession class that parses
+all server inputs and/or outputs and passes a dict containing the parsed
+metadata to a method of your choosing.
+
+This is useful for recording player activity where necessary for auditing,
+usage data or post-incident forensic analysis.
+
+Somewhere in your game (or even right there in settings.py) should be a 
+method that receives dict objects and does something with them, whether 
+writing to a database, a file, or some external queue. A simplistic example 
+might be something as brief as:
+
+from django.utils import timezone
+from evennia.utils.logger import log_file
+import json
+
+def just_write_it(data, **kwargs):
```

Comment:
The settings generally holds strings. Adding functional code directly to settings is not a good way to do it and shouldn't be recommended. The way Evennia usually handles this is to supply the path (as a string) to the method and then use one of the utilities in `evennia.utils.utils` to load it. To this end this minimal output method (`just_write_it`) should be included in this contrib as a default if you don't supply your own method - not only does it make it easier to test, it also makes for a more complete contrib. Ideally the "scrubbing" mentioned later in this docstring should be handled with the assumption of the default cmdset (alternatively log a warning about it to console).

Reply:
Agreed; I will break the example out of settings and remove the suggestion to write functions in settings. I had the example in the contrib initially but assumed since it wasn't directly related to the serversession object I'd be asked to remove it.

I like the idea of making the example scrub the default cmdset. I think it's an excellent compromise given the concerns you also brought up in the previous PR.